graph.go

```diff
@@ -40,23 +40,30 @@ import (
 // Opts captures customizable functionality like logging
 type Opts struct {
 	CustomLogger func(args ...interface{})
+	Decorator    func(b Builder) Builder
```

Comment:
Shouldn't this be like `func(ctx context.Context, b graph.Builder) (string, error) ` and the client will call the update instead the library?